blocks/image-placeholder/index.js

```diff
@@ -23,15 +23,15 @@ import { rawHandler } from '../api';
  *
  * @return {Object} Rendered placeholder.
  */
-export default function ImagePlaceholder( { className, icon, label, onSelectImage, multiple = false } ) {
+export default function ImagePlaceholder( { className, icon, label, onSelectImage, multiple = false, parentId = null } ) {
```

Comment:
@gziolo @aduth Could I get some direction on how `parentId` is meant to be passed into `ImagePlaceholder`?

I see `ImagePlaceholder` is used by the following blocks: `cover-image`, `gallery`, `image`. However, it appears these blocks don't know of their post context. It seems like the "best" way to pass their post context is as some magic attribute. However, it doesn't seem like this is better than `ImagePlaceholder` referencing some magic global variable.

Reply:
> It seems like you can use `withSelect` to inject it here.

Can you provide some direction on how this would work?

Reply:
![screen shot 2018-04-16 at 09 45 11](https://user-images.githubusercontent.com/699132/38796182-e424a0b2-415a-11e8-8587-b3c9b220c9a6.png)

So in this case:
```js
export default withSelect(
    ( select) => {
        parentId: select( 'core/editor' ).getCurrentPostId(),
    }, 
)( ImagePlaceholder );
```

Reply:
Updated in a4ed758e